src/Modules/Workspaces/Commands.Workspaces/GetPowerBIWorkspace.cs

```diff
@@ -125,5 +147,36 @@ public override void ExecuteCmdlet()
                 this.Logger.WriteObject(workspaces, true);
             }
         }
+
+        private void ExecuteCmdletWithAll()
+        {
+            var allWorkspaces = new List<Workspace>();
+            using (var client = this.CreateClient())
+            {
+                var top = 5000;
+                var skip = 0;
+                var workspacesCountBeforeApiCall = 0L;
+                var workspacesCountAfterApiCall = 0L;
+
+                // GetWorkspacesAsAdmin is called with 5000 as the top which retrieves 5000 workspaces in each call.
+                // Loop will terminate when the workspaces retrieved are less than 5000.
+                do
+                {
+                    var result = client.Workspaces.GetWorkspacesAsAdmin(expand: "users", filter: this.Filter, top: top, skip: skip);
+                    allWorkspaces.AddRange(result);
+                    workspacesCountBeforeApiCall = workspacesCountAfterApiCall;
+                    workspacesCountAfterApiCall = allWorkspaces.Count;
```

Comment:
Is .Count total count or count that was retrieved?

Reply:
That is the count of the total number of workspaces retrieved so far. But it is not longer needed since I switched it to use just the result.Count()